frontend/src/main/scala/bloop/data/Project.scala

```diff
@@ -76,6 +85,30 @@ final case class Project(
     customWorkingDirectory.getOrElse(baseDirectory)
   }
 
+  /** Returns concatenated list of "sources" and expanded "sourcesGlobs". */
+  def allSourceFilesAndDirectories(): List[AbsolutePath] = {
```

Comment:
Should we remove the extra `()`?

Reply:
Removed. I added the parentheses to indicate it's a stateful method (it returns different values depending on I/O) on an otherwise stateless case class. Now that it returns a `Task` it's stateless :)